packages/ipfs/src/index.js

```diff
@@ -2,4 +2,8 @@
 
 const IPFS = require('ipfs-core')
```

Comment:
I think it would be simpler to do this instead

```suggestion
/**
 * @typedef {import(‘ipfs-core/src/components’)} IPFS
 */
 
 module.exports = require('ipfs-core')
```

Reply:
It does not work because `@typedef {import('ipfs-core/src/components')} IPFS` will not overlap onto the exports

Generated `d.ts` file:

```ts
declare const _exports: typeof import('ipfs-core/src')
export = _exports;
export type IPFS = import('ipfs-core/src/components');
```

We can't use 
```ts
import type IPFS from "ipfs"
```

Reply:
@Xmader I realize that, but point I was trying to make is, having export type `IPFS` is just simpler and in the nutshell will give us:

```ts
import { IPFS, create } from "ipfs"
```

Furthermore I expect that as we adopt approach proposed in #3413 we might end up with `IPFS` interface that in the future could be exported in place of `IPFS` type.

That said, it's an @achingbrain call